### PR TITLE
[WIP] enable DistriOptimizerV2

### PR DESCRIPTION
**Pull request by Le-Zheng**
To switch DistriOptimizerVersion:
Python examples: 
`from zoo.common.nncontext import set_optimizer_version `
`set_optimizer_version("optimizerv2")`
Scala examples:
`import com.intel.analytics.zoo.common.NNContext`
`NNContext.setOptimizerVersion("optimizerv2")`

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 1]
```diff
@@ -1489,12 +1520,218 @@ private[zoo] class InternalDistriOptimizer[T: ClassTag] (
...
+    val sc = validateRDD.sparkContext
+
+    val coresPerNode = EngineRef.getCoreNumber()
+    val _subModelNumber = EngineRef.getEngineType() match {
+      case MklBlas => coresPerNode
+      case MklDnn => 1
+      case _ => throw new IllegalArgumentException
+    }
+
+    val models = if (null != cachedModels) {
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala:1669]: get models from distrioptimizerV2

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 2]
```diff
@@ -1640,3 +1877,80 @@ object InternalDistriOptimizer {
     trainingModel
   }
 }
+
+object InternalDistriOptimizerV2 {
+  val logger = Logger.getLogger(this.getClass)
+  
+  protected def validate[T](validationFeatureSet: FeatureSet[MiniBatch[T]],
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala:1884]: different from v1?

Comment (Le-Zheng) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala:1845]: It's the same. But the Cache type is different.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala (hunk 1 above)
Comment (Le-Zheng) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: Done. Use `val models = this.cachedModels`

**Comment by Le-Zheng** [comment]
Too many conflicts in this pr. Have opened new pr https://github.com/intel-analytics/analytics-zoo/pull/2791.

**Pull request closed without merging**